src/Database/index.js

```diff
@@ -55,11 +55,18 @@ export default class Database {
       }
 
       invariant(
-        !record._isCommitted || record._hasPendingUpdate,
+        !record._isCommitted || record._hasPendingUpdate || record._hasPendingDelete,
         `Cannot batch a record that doesn't have a prepared create or prepared update`,
       )
 
-      if (record._hasPendingUpdate) {
+      // Deletes take presedence over updates
+      if (record._hasPendingDelete !== false) {
+        if (record._hasPendingDelete === 'destroy') {
+          return ops.concat([['destroyPermanently', record]])
+        }
+        
+        return ops.concat([['markAsDeleted', record]])
```

Comment:
you can replace the 5 lines above with just one (with a ternary record._hasPendingDelete === 'destroy' ? 'destroy...' : 'markAsDeleted') ?